lib/src/typewriter.dart

```diff
@@ -61,8 +66,18 @@ class TypewriterAnimatedText extends AnimatedText {
   /// Widget showing partial text
   @override
   Widget animatedBuilder(BuildContext context, Widget child) {
+    /// Output of CurveTween is in the range [0, 1] for majority of the curves.
+    /// It is converted to [0, textCharacters.length].
     final textLen = textCharacters.length;
-    final typewriterValue = _typewriterText.value;
+    var typewriterValue =
+        (_typewriterText.value * textCharacters.length).round();
+
+    if (typewriterValue < 0) {
```

Comment:
How could `typewriterValue` be less than 0?  (This _if_ seems unnecessary.)

Reply:
I guess the example I gave above is applicable here as well.

Reply:
Got it.  However, this could also be resolved with the `min/max` expression.

Reply:
Don't forget the `extraLengthForBlinks` adjustment:
```dart
final typewriterValue = (min(max(0, _typewriterText.value), 1.0)
  * (textCharacters.length + extraLengthForBlinks)).round();
```